include/bouncer.h

```diff
@@ -335,6 +336,7 @@ struct PgUser {
 struct PgDatabase {
 	struct List head;
 	char name[MAX_DBNAME];	/* db name for clients */
+	struct AATree user_passwds;	/* user passwords for this database from auth_query */
```

Comment:
A hashtable to store user passwords is more ideal but is more difficult (would have to pull in glibc).